daq/base_gateway.py

```diff
@@ -147,14 +150,14 @@ def release_test_port(self, test_port):
         self.test_ports.remove(test_port)
 
     def _switch_port(self, offset):
-        return self.port_set * self.SET_SPACING + offset
+        return self.port_set * self._gw_set_size + offset
 
     def _is_target_expected(self, target):
         if not target:
             return False
         target_mac = target['mac']
         if target_mac in self.targets:
-            return True
+            return target_mac
```

Comment:
Looking at the code, it is not immediately clear to me why this change is needed. Based on hoqw it is used, it looks like it does expect it to be a bool.